features/prog_test.go

```diff
@@ -0,0 +1,85 @@
+package features
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"testing"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/internal/testutils"
+)
+
+var progTypeMinVersion = map[ebpf.ProgramType]string{
+	ebpf.SocketFilter:          "3.19",
+	ebpf.Kprobe:                "4.1",
+	ebpf.SchedCLS:              "4.1",
+	ebpf.SchedACT:              "4.1",
+	ebpf.TracePoint:            "4.7",
+	ebpf.XDP:                   "4.8",
+	ebpf.PerfEvent:             "4.9",
+	ebpf.CGroupSKB:             "4.10",
+	ebpf.CGroupSock:            "4.10",
+	ebpf.LWTIn:                 "4.10",
+	ebpf.LWTOut:                "4.10",
+	ebpf.LWTXmit:               "4.10",
+	ebpf.SockOps:               "4.13",
+	ebpf.SkSKB:                 "4.14",
+	ebpf.CGroupDevice:          "4.15",
+	ebpf.SkMsg:                 "4.17",
+	ebpf.RawTracepoint:         "4.17",
+	ebpf.CGroupSockAddr:        "4.17",
+	ebpf.LWTSeg6Local:          "4.18",
+	ebpf.LircMode2:             "4.18",
+	ebpf.SkReuseport:           "4.19",
+	ebpf.FlowDissector:         "4.20",
+	ebpf.CGroupSysctl:          "5.2",
+	ebpf.RawTracepointWritable: "5.2",
+	ebpf.CGroupSockopt:         "5.3",
+	ebpf.Tracing:               "5.5",
+	ebpf.StructOps:             "5.6",
+	ebpf.Extension:             "5.6",
+	ebpf.LSM:                   "5.7",
+	ebpf.SkLookup:              "5.9",
+}
+
+func TestHaveProgType(t *testing.T) {
```

Comment:
This could be run with `t.Parallel()`.

Reply:
Done. Had to make use of some IMO not so elegant variable copying as mentioned [here](https://gist.github.com/posener/92a55c4cd441fc5e5e85f27bca008721#how-to-solve-this), but that works.

Reply:
This is common in Go. :slightly_smiling_face: Good catch!